distribution/test/itests/test-itests-common/src/main/java/org/codice/ddf/itests/common/config/ConfigureTestCommons.java

```diff
@@ -13,69 +13,101 @@
-    properties.put("attributeMap", securityAttributeMappings);
-    config.update(properties);
+  public static Dictionary<String, Object> configureService(
+      String pid, Dictionary<String, Object> props, AdminConfig adminConfig) throws IOException {
+    Configuration config = adminConfig.getConfiguration(pid, null);
+    Dictionary<String, Object> oldProps = config.getProperties();
+    config.update(props);
+
+    // verify update
+    with()
+        .pollInterval(1, SECONDS)
+        .await()
+        .atMost(TimeUnit.MINUTES.toSeconds(2), SECONDS)
+        .ignoreExceptions()
+        .until(
+            () -> {
+              Dictionary<String, Object> savedProps = config.getProperties();
+              List<String> newKeys = Collections.list(props.keys());
+              for (String key : newKeys) {
+                Object savedValue = savedProps.get(key);
+                Object incomingValue = props.get(key);
+                if (incomingValue.getClass().isArray() && savedValue.getClass().isArray()) {
+                  if (!Arrays.equals((Object[]) savedValue, (Object[]) incomingValue)) {
+                    return false;
+                  }
+                } else if (!savedValue.equals(incomingValue)) {
+                  return false;
+                }
+              }
+              return true;
+            });
+
+    return oldProps;
   }
 
-  public static void configureFilterInvalidMetacards(
-      String filterErrors, String filterWarnings, AdminConfig configAdmin) throws IOException {
-    Configuration config =
-        configAdmin.getConfiguration(METACARD_VALIDITITY_FILTER_PLUGIN_SERVICE_PID, null);
-
+  public static Dictionary<String, Object> configureMetacardValidityFilterPlugin(
+      List<String> securityAttributeMappings,
+      boolean filterErrors,
+      boolean filterWarnings,
+      AdminConfig configAdmin)
+      throws IOException {
     Dictionary<String, Object> properties = new DictionaryMap<>();
-    properties.put("filterErrors", filterErrors);
-    properties.put("filterWarnings", filterWarnings);
-    config.update(properties);
+    properties.put("attributeMap", securityAttributeMappings);
+    properties.put("filterErrors", String.valueOf(filterErrors));
+    properties.put("filterWarnings", String.valueOf(filterWarnings));
+    return configureMetacardValidityFilterPlugin(properties, configAdmin);
   }
 
-  public static void configureEnforceValidityErrorsAndWarnings(
-      String enforceErrors, String enforceWarnings, AdminConfig configAdmin) throws IOException {
-    Configuration config =
-        configAdmin.getConfiguration(METACARD_VALIDITITY_MARKER_PLUGIN_SERVICE_PID, null);
-
-    Dictionary<String, Object> properties = new DictionaryMap<>();
-    properties.put("enforceErrors", enforceErrors);
-    properties.put("enforceWarnings", enforceWarnings);
-    config.update(properties);
+  public static Dictionary<String, Object> configureMetacardValidityFilterPlugin(
```

Comment:
❓ If only one field is passed in, the others in the config are nulled out right? Do we still want to allow this to happen by passing in a dictionary, or should we force them to configure every field? Could be a case by case basis for each config.

Reply:
That's a good point. I was mainly intended these ones that take a dictionary to be used to reset the former configuration that was returned as a dictionary when the other method is called. Maybe I should add a comment to that effect for what the intent is